Explain how to adjust the home directory (#7759)
jenkins-infra/jenkins.io: content/doc/developer/testing/index.adoc

----

==== Customizing the `JENKINS_HOME` Directory

> If you want to adjust the `JENKINS_HOME` within a test, you can make use of the `JenkinsRule`-methods `withExistingHome()` or `withNewHome()`.

Reviewer: These methods seem to be rarely used.  A GitHub search showed only two usages in the jenkinsci organization.

* [LogStash plugin](https://github.com/jenkinsci/logstash-plugin/blob/a10e30d246c9b51919c4e741d915f23fc7353367/src/test/java/jenkins/plugins/logstash/LogstashBuildWrapperConversionTest.java#L29-L32)
* [LDAP email plugin](https://github.com/jenkinsci/ldapemail-plugin/blob/05051099ee9f56a530ca82d1ef5395c07026737d/src/test/java/com/mtvi/plateng/hudson/ldap/GoodConfigLoadTest.java#L26-L29)

A minor phrasing change.  I'm not sure of the preferred replacement for these test methods, but I suspect that there are preferred replacements..

```suggestion
If you want to adjust the `JENKINS_HOME` within a test, use the `JenkinsRule`-methods link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html#withExistingHome(java.io.File)[`withExistingHome()`] or link:https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/JenkinsRule.html#withNewHome()[`withNewHome()`].
```

> Let's assume in the directory "src/test/resources/home" everything is prepared for a good test suite, and you want to use that.

Reviewer 2: ```suggestion
Let's assume in the directory "src/test/resources/home" everything is prepared for a good test suite, which you should use.
```

> Let's assume in the directory "src/test/resources/home" everything is prepared for a good test suite, which you should use.

Reviewer 3: ```suggestion
Let's assume everything in the directory "src/test/resources/home" is prepared for a good test suite, which you should use.
```
Small wording placement change to help with readability.

This can be achieved with the following example:

[source,java]
----

    @Rule
    public JenkinsRule j = new JenkinsRule().withExistingHome(new File("src/test/resources/home"));
----

> If you need a fresh `JENKINS_HOME` you can use `withNewHome()` which creates a fresh home, what the default of the `JenkinsRule()` at the beginning is.

Reviewer 2: ```suggestion
If you need a fresh `JENKINS_HOME` you can use `withNewHome()` which creates a fresh home, which is by default what the `JenkinsRule()` at the beginning is.
```

> If you need a fresh `JENKINS_HOME` you can use `withNewHome()` which creates a fresh home, which is by default what the `JenkinsRule()` at the beginning is.

Reviewer 3: ```suggestion
If you need a fresh `JENKINS_HOME`, you can use `withNewHome()` to create a fresh home, which `JenkinsRule()` at the beginning defaults to.
```
Small wording placement/adjustment for readability and to clarify the statement.
